functions/helloworld/test/sample.integration.storage.test.js

```diff
@@ -14,69 +14,100 @@
  */
 
 // [START functions_storage_integration_test]
-const childProcess = require('child_process');
 const assert = require('assert');
+const delay = require('delay');
+const execPromise = require('child-process-promise').exec;
+const path = require('path');
+const supertest = require('supertest');
 const uuid = require('uuid');
 
-it('helloGCS: should print uploaded message', done => {
-  const startTime = new Date(Date.now()).toISOString();
+const request = supertest(process.env.BASE_URL || 'http://localhost:8080');
+const cwd = path.join(__dirname, '..');
+
+it('helloGCS: should print uploaded message', async () => {
   const filename = uuid.v4(); // Use a unique filename to avoid conflicts
 
-  // Mock GCS call, as the emulator doesn't listen to GCS buckets
-  const data = JSON.stringify({
-    name: filename,
-    resourceState: 'exists',
-    metageneration: '1',
-  });
-
-  childProcess.execSync(`functions-emulator call helloGCS --data '${data}'`);
-
-  // Check the emulator's logs
-  const logs = childProcess
-    .execSync(`functions-emulator logs read helloGCS --start-time ${startTime}`)
-    .toString();
-  assert.ok(logs.includes(`File ${filename} uploaded.`));
-  done();
+  const data = {
+    data: {
+      name: filename,
+      resourceState: 'exists',
+      metageneration: '1',
+    },
+  };
+
+  const proc = execPromise(
+    `functions-framework --target=helloGCS --signature-type=event`,
+    {timeout: 800, shell: true, cwd: cwd}
+  );
+
+  await delay(600);
```

Comment:
Why the delay instead of working once the `execPromise` resolves?

Reply:
IIUC, `execPromise` only resolves once the process is terminated. Because sending a request should occur _before_ the process terminates, we have to use a `delay` instead.

Reply:
+1. This may produce flaky tests if kept.

Reply:
What about:
- Filing a bug against the FF if it isn't operational immediately.
- Retrying the request rather than delaying an arbitrary number of ms.

Reply:
I'll look into the second option. `retry-axios` might be a good fit for this (though we may not be able to use `supertest` without [something like this](https://github.com/segmentio/superagent-retry)).

Reply:
Heads up that `axios` isn't actively maintained anymore. We should take that into consideration if we decide to switch to `axios`.